nifpga/session.py

```diff
@@ -390,8 +410,272 @@ def read(self):
+        read_func = self._get_read_function()
+        read_func(self._session, self._resource, buf, len(self))
+        val = [bool(elem) if self._datatype is DataType.Bool else elem for elem in buf]
+        return val
+
+    def _get_read_function(self):
+        """ Sets this registers read function based on the data type from the
+        bitfile."""
+        return self._nifpga["ReadArray%s" % self.datatype]
+
+    def _get_write_function(self):
+        """ Sets this registers read function based on the data type from the
+        bitfile."""
+        return self._nifpga["WriteArray%s" % self.datatype]
+
+
+class _FxpRegister(_ArrayRegister):
+    """
+    _FxpRegister is a private class that does all the work of converting the
+    LabVIEW fixed point data type into something more native to python. This
+    implementation will return the Decimal type when calling read(), but should
+    be able to handle any native python data types when writing.
+
+    Just like the other Registers we do not support users creating their
+    instances of _FxpRegister, but all of the fixed point registers from a
+    given bitfile will exist under the sessions.register property.
+
+    Fixed point registers should be easily used from python with a few caveats:
+        1. Trying to write data that does not conform to boundaries of the
+        defined register, will be coerced and the user will be warned. This is
+        either data outside the maximum and minimum values, or data that is not
+        a multiple of the delta.
+        2. Using Read() and Write() will only return the numerically values of
+        a fixed point register. In order to access the overflow status bit,
+        users must call the overflow attribute on the Register. The register
+        will only have a set overflow status given, the Register has been
+        enabled, and has at least been read from or written to before.
+    """
+    def __init__(self,
+                 session,
+                 nifpga,
+                 bitfile_register,
+                 base_address_on_device):
+        super(_FxpRegister, self).__init__(session,
+                                           nifpga,
+                                           bitfile_register,
+                                           base_address_on_device)
+
+        self._word_length = bitfile_register.word_length
+        self._integer_word_length = bitfile_register.integer_word_length
+        self._radix_point = self._calculate_radix_point()
+        self._overflow_enabled = bitfile_register.overflow
+        self._signed = bitfile_register.signed
+        self._delta = self._calculate_delta(bitfile_register.word_length,
+                                            bitfile_register.integer_word_length)
+
+    @property
+    def overflow(self):
+        """ This property should only set be whenever this register has overflow
+        enabled"""
+        return self._overflow
```

Comment:
It seems a little odd to have overflow be a separate property on the Register.  Overflow is transferred on every read and write right?  Separating out overflow from read and write is less multi-thread friendly.  For registers that support overflow, I was expecting something like:

(data, overflow) = read()
write(data, overflow)

What do you think?

Reply:
Fixed in commit "Refactor overflow bit"